packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -0,0 +1,1274 @@
+///
+/// When open, the [CupertinoContextMenu] shows the child, or the widget returned
+/// by [previewBuilder] if given, in a large full-screen [Overlay] with a list
+/// of buttons specified by [actions]. The child/preview is placed in an
+/// [Expanded] widget so that it will grow to fill the Overlay if its size is
+/// unconstrained.
+///
+/// When closed, the CupertinoContextMenu simply displays the child as if the
+/// CupertinoContextMenu were not there. Sizing and positioning is unaffected.
+/// The menu can be closed like other [PopupRoute]s, such as by tapping the
+/// background or by calling `Navigator.pop(context)`. Unlike PopupRoute, it can
+/// also be closed by swiping downwards.
+///
+/// The [previewBuilder] parameter is most commonly used to display a slight
+/// variation of [child]. See [previewBuilder] for an example of rounding the
+/// child's corners and allowing its aspect ratio to expand, similar to the
+/// Photos app on iOS.
+///
+/// {@tool dartpad --template=stateless_widget_material}
+///
+/// This sample shows a very simple CupertinoContextMenu for an empty red
+/// 100x100 Container. Simply long press on it to open.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     key: scaffoldKey,
+///     body: Center(
+///       child: Container(
+///         width: 100,
+///         height: 100,
+///         child: CupertinoContextMenu(
+///           child: Container(
+///             color: Colors.red,
+///           ),
+///           actions: <CupertinoContextMenuAction>[
+///             CupertinoContextMenuAction(
+///               child: const Text('Action one'),
+///               onPressed: () {
+///                 Navigator.pop(context);
+///               },
+///             ),
+///             CupertinoContextMenuAction(
+///               child: const Text('Action two'),
+///               onPressed: () {
+///                 Navigator.pop(context);
+///               },
+///             ),
+///           ],
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [Apple's HIG for Context Menus](https://developer.apple.com/design/human-interface-guidelines/ios/controls/context-menus/)
+class CupertinoContextMenu extends StatefulWidget {
+  /// Create a context menu.
+  ///
+  /// [actions] is required and cannot be null. It must contain at least one
+  /// [CupertinoContextMenuAction].
+  ///
+  /// [child] is required and cannot be null.
+  CupertinoContextMenu({
+    Key key,
+    @required this.actions,
+    @required this.child,
+    this.previewBuilder,
+  }) : assert(actions != null && actions.isNotEmpty),
+       assert(child != null),
+       super(key: key);
+
+  /// The widget that can be "opened" with the [CupertinoContextMenu].
+  ///
+  /// When the [CupertinoContextMenu] is long-pressed, the menu will open and
+  /// this widget (or the widget returned by [previewBuilder], if provided) will
```

Comment:
I think this may be revealing a detail that's not important: that the child/previewBuilder will be moved to the overlay. Can we explain what it means to "open" the menu just in terms of the route we're going to push? That approach jibes with using Navigator.pop() to close the menu.